jubilant/_juju.py

```diff
@@ -205,6 +205,151 @@ def add_unit(
 
         self.cli(*args)
 
+    def bootstrap(
+        self,
+        cloud: str | None = None,
+        controller: str | None = None,
+        *,
+        add_model: str | None = None,
+        agent_version: str | None = None,
+        auto_upgrade: bool = False,
+        bootstrap_base: str | None = None,
+        bootstrap_constraints: Mapping[str, str] | None = None,
+        bootstrap_image: str | None = None,
+        bootstrap_series: str | None = None,
+        build_agent: bool = False,
+        clouds: bool = False,
+        config: Mapping[str, ConfigValue] | None = None,
+        constraints: Mapping[str, str] | None = None,
+        controller_charm_channel: str | None = None,
+        controller_charm_path: str | pathlib.Path | None = None,
+        credential: str | None = None,
+        db_snap: str | pathlib.Path | None = None,
+        db_snap_asserts: str | pathlib.Path | None = None,
+        force: bool = False,
+        keep_broken: bool = False,
+        metadata_source: str | pathlib.Path | None = None,
+        model_default: Mapping[str, ConfigValue] | None = None,
+        no_switch: bool = False,
+        regions: str | None = None,
+        storage_pool: Mapping[str, str] | None = None,
+        to: str | None = None,
+    ) -> str:
```

Comment:
I think we won't need many of these arguments, and we should trim it down a bit. I suggest getting rid of the ones having to do with customising/developing the controller code:

- agent_version
- auto_upgrade
- bootstrap_image
- build_agent 
- controller_charm_channel
- controller_charm_path
- db_snap
- db_snap_asserts
- metadata_source

In addition, I think we should drop these:

- clouds (this is a completely different mode, and should be done via `juju clouds --all` or similar instead; let's just drop it here)
- regions (similar to *clouds*)
- keep_broken: I don't think this one will be useful in an automated context